### PR TITLE
New viewing party

### PR DESCRIPTION
**Pull request by EmmyMorris**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Rubocop has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A
Can not add new view parties or see parties on a dashboard.

## What is the new behavior?

-Can create a new view party
-Can see hosted parties
-Can see parties invited to

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: app/controllers/parties_controller.rb
Comment (mkrumholz) [inline app/controllers/parties_controller.rb:7]: ⛏ Nitpick: it is conventional to try not to send more than 1 instance variable through the controller. Is there a prettier way to do this? Maybe we could consider sending those params all as a hash or possibly creating a movie object that we could send through to the view?

File: app/controllers/parties_controller.rb
Comment (mkrumholz) [inline app/controllers/parties_controller.rb]: This method is pretty gigantic. It does a ton of different things. I think we should consider creating some helper methods to deal with all this logic. Happy to talk through this, but here are a couple of thoughts:
1. One method that deals with runtime being too long/short
2. One method dealing with no friends being present (once that's fixed and tested)
3. A method that deals with creating invitations
4. etc.

File: app/controllers/parties_controller.rb
Comment (mkrumholz) [inline app/controllers/parties_controller.rb]: 🛑 Critical: I don't think this is working. I tried it out on local and it let me create a party without any friends. We should consider either removing this requirement (and adding an edit function, maybe?) or enforcing it and make sure we're testing for it.

File: app/controllers/parties_controller.rb
Comment (mkrumholz) [inline app/controllers/parties_controller.rb]: ⛏ Nitpick: I might split this line up so it's not so long like:
```rb
params.require(:party)
  .permit(:movie_title, :duration, :date, :start_time)
  .merge({ external_movie_id: params[:external_movie_id] })
```

File: app/views/dashboard/show.html.erb
Comment (mkrumholz) [inline app/views/dashboard/show.html.erb:16]: 🎉 Nice add here! Loved these little messages for mini-sad paths

File: app/views/dashboard/show.html.erb
Comment (mkrumholz) [inline app/views/dashboard/show.html.erb]: 💡 We may want to put these each on a new line or find a creative way to display them, so they don't all end up in one block of friends.

File: app/views/dashboard/show.html.erb
Comment (mkrumholz) [inline app/views/dashboard/show.html.erb:49]: Same here as above ^

File: app/views/movies/show.html.erb
Comment (mkrumholz) [inline app/views/movies/show.html.erb:2]: Re: variables in the controller -- maybe we should consider sending this as one hash like {movie: {title: ..., runtime: ..., etc. }?

File: app/views/parties/new.html.erb
Comment (mkrumholz) [inline app/views/parties/new.html.erb:26]: 🎉 Awesome work figuring out this collection check boxes thing, as well as sending the external movie id as a hidden field!

File: config/routes.rb
Comment (mkrumholz) [inline config/routes.rb]: ⛏These should probably get removed!

File: db/migrate/20210712201254_change_viewing_parties_to_parties.rb
Comment (mkrumholz) [inline db/migrate/20210712201254_change_viewing_parties_to_parties.rb:5]: 👏🏻 Awesome job with alllll these migrations!

File: spec/features/parties/dashboard_party_spec.rb
Comment (mkrumholz) [inline spec/features/parties/dashboard_party_spec.rb:21]: 💡I wonder if we should add FactoryBot to help clean up some of our testing. Could be a nice refactor at the end, if desired. Definitely not critical but could avoid us dealing with passwords and all that every time.

File: spec/features/parties/dashboard_party_spec.rb
Comment (mkrumholz) [inline spec/features/parties/dashboard_party_spec.rb:25]: ⛏ Realizing here that we technically seem to let users create parties where the time is stored as a Timestamp for a different date than the date of the party. I think this could be a reason why I might have chosen to store these as one datetime object. Maybe we could refactor into that if we have time?

File: spec/features/parties/dashboard_party_spec.rb
Comment (mkrumholz) [inline spec/features/parties/dashboard_party_spec.rb:105]: ❓ I wonder if this should be a separate test?

File: spec/features/parties/dashboard_party_spec.rb
Comment (mkrumholz) [inline spec/features/parties/dashboard_party_spec.rb:177]: Kinda same here. We could do a context block for when the user is invited vs hosting, maybe? Seems like these are each 2 tests in one.

File: spec/features/parties/dashboard_party_spec.rb
Comment (mkrumholz) [inline spec/features/parties/dashboard_party_spec.rb:2]: ❓ Do we have any testing for the sad paths on this dashboard view for parties? I saw that you added some messages for those sad paths, but I'm not seeing tests for them. Like, maybe we should add a test that covers when they're not invited to anything/not hosting anything?

File: spec/features/parties/new_party_spec.rb
Comment (mkrumholz) [inline spec/features/parties/new_party_spec.rb]: 🤔 Interestingly, this seems to be passing, but I was able to create a party with no friends on my local. I will try to do some more testing on this to be certain.

File: spec/features/parties/new_party_spec.rb
Comment (mkrumholz) [inline spec/features/parties/new_party_spec.rb]: ❓ Should we maybe add some tests to ensure the display is correct for date and time?

**Review by mkrumholz — CHANGES_REQUESTED** [review]

File: app/controllers/parties_controller.rb
Comment (EmmyMorris) [reply app/controllers/parties_controller.rb]: Oh no! I forgot to test this after running rubocop, it was a correction it suggested to me!

File: spec/features/parties/dashboard_party_spec.rb
Comment (EmmyMorris) [reply spec/features/parties/dashboard_party_spec.rb:21]: That would be fun to try! Ive never used FactoryBot before and would love to try!

File: app/controllers/parties_controller.rb
Comment (mkrumholz) [reply app/controllers/parties_controller.rb:7]: Just noting here again after our convo that we should wait on this until my piece is merged in and we can maybe refactor to use objects instead of constantly passing params

File: spec/features/parties/new_party_spec.rb
Comment (mkrumholz) [reply spec/features/parties/new_party_spec.rb]: Ahh was hitting the other error. Looks better now!

File: app/controllers/parties_controller.rb
Comment (mkrumholz) [inline app/controllers/parties_controller.rb:25]: This looks way better, but definitely still some opportunity for future refactoring to clean up the controller a bit.

**Review by mkrumholz — APPROVED** [review]
Nice work on refactoring and making those updates! Looking lots closer and after we merge in my branch as well, I think those later refactors will go pretty smoothly.

**Pull request merged**
